### PR TITLE
Bjorncs/jdisc security filters

### PR DESCRIPTION
**Pull request by bjorncs**

### CODE REVIEW

**Review by gjoranv — APPROVED** [review]

**Comment by gjoranv** [comment]
Please ignore my approval for now, it was sent prematurely by accident.

File: jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filters/cors/CorsSecurityRequestFilterBase.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import com.yahoo.jdisc.http.filter.SecurityRequestFilter;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.yahoo.jdisc.http.filters.cors.CorsLogic.createCorsResponseHeaders;
+
+/**
+ * Security request filters should extend this base class to ensure that CORS header are included in the response of a rejected request.
```
Comment (gjoranv) [inline jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filters/cors/CorsSecurityRequestFilterBase.java]: s/header/headers/

**Comment by bjorncs** [comment]
@gjoranv PTAL. See last commit for the renaming of Java package and class names.

File: jdisc-security-filters/src/test/java/com/yahoo/jdisc/http/filters/cors/CorsSecurityRequestFilterBaseTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author bjorncs
+ */
+public class CorsSecurityRequestFilterBaseTest {
+
+    @Test
+    public void adds_cors_headers_when_filter_reject_request() {
```
Comment (gjoranv) [inline jdisc-security-filters/src/test/java/com/yahoo/jdisc/http/filters/cors/CorsSecurityRequestFilterBaseTest.java]: Please remember to adhere to the implicit "require that" naming. For example, `cors_headers_are added_when_filter_rejects_request`

File: jdisc-security-filters/src/test/java/com/yahoo/jdisc/http/filters/cors/CorsSecurityRequestFilterBaseTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            this.errorResponse = new ErrorResponse(statusCode, message);
+        }
+
+        @Override
+        protected Optional<ErrorResponse> filter(DiscFilterRequest request) {
+            return Optional.ofNullable(this.errorResponse);
+        }
+    }
+
+}
```
Comment (gjoranv) [inline jdisc-security-filters/src/test/java/com/yahoo/jdisc/http/filters/cors/CorsSecurityRequestFilterBaseTest.java]: Please remember those trailing newlines. (I think you have said the same to me sometimes.) 😉

File: jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filters/cors/CorsLogic.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.jdisc.http.filters.cors;
```
Comment (gjoranv) [inline jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filters/cors/CorsLogic.java]: I suggest to use the package name `com.yahoo.jdisc.filter.security.cors`.

File: jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filters/cors/CorsPreflightSecurityRequestFilter.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+ * Runs after all standard security request filters, but before BouncerFilter, as the browser does not send
+ * credentials with pre-flight requests.
+ * </p>
+ *
+ * @author andreer
+ * @author gv
+ * @author bjorncs
+ */
+@Provides("CorsPreflightSecurityRequestFilter")
+public class CorsPreflightSecurityRequestFilter implements SecurityRequestFilter {
```
Comment (gjoranv) [inline jdisc-security-filters/src/main/java/com/yahoo/jdisc/http/filters/cors/CorsPreflightSecurityRequestFilter.java]: Using the proposed package name with 'security', you can shorten the filter class names.

**Review by gjoranv — CHANGES_REQUESTED** [review]
See inline comments.

**Review by gjoranv — CHANGES_REQUESTED** [review]
Good, just remove 'http' from the package name, and we're fine.

**Comment by bjorncs** [comment]
I think we should keep `http` in the package name to match the existing types in `jdisc_http_service` and other similar modules.

**Comment by gjoranv** [comment]
ok, agreed

**Review by gjoranv — APPROVED** [review]

**Pull request merged**